packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -4,28 +4,33 @@
  */
 
 import { QueryConjunctions } from '@tupaia/server-boilerplate';
+import { getSortByKey } from '@tupaia/utils';
 
-import { Writable, ObjectLikeKeys, Flatten } from '../../../types';
-import { EntityFilter, EntityFields } from '../../../models';
+import { Writable, NumericKeys, ObjectLikeKeys, Flatten } from '../../../types';
```

Comment:
@rohan-bes  I had to add/refactor a bunch of code because `generational_distance` is the first non-string field here...

I could be a bit lazy and define `generational_distance` as a string - the code wouldn't complain :wink: but it felt wrong...

That being said, I believe that simplifying the types in this file would make it more manageable. They are very cool, but they go a long way to ensure that things are DRY.

As an example, instead of inferring the nested JSON fields and their types from `EntityFields`, they could be explicitly defined as

```ts
interface FilterableFields {
  code: 'name';
  // ...etc
  atttributes_type: 'string';
  generational_distance: number;
}
```

Tests could be used to assert that at least the types used are compatible with the DB schema.

Reply:
In Grug we trust 😄